utils/src/main/java/com/cloud/utils/net/NetUtils.java

```diff
@@ -775,11 +776,19 @@ public static String getSubNet(final String ip, final String netmask) {
     }
 
     public static String getCidrSubNet(final String ip, final long cidrSize) {
-        final long numericNetmask = 0xffffffff >> MAX_CIDR - cidrSize << MAX_CIDR - cidrSize;
+        final long numericNetmask = netMaskFromCidr(cidrSize);
         final String netmask = NetUtils.long2Ip(numericNetmask);
         return getSubNet(ip, netmask);
     }
 
+    /**
+     * @param cidrSize
+     * @return
+     */
+    static long netMaskFromCidr(final long cidrSize) {
+        return ((long)0xffffffff) >> MAX_CIDR - cidrSize << MAX_CIDR - cidrSize;
+    }
+
```

Comment:
I think that it is fine letting the method with default modifier. 
The point of a test case is that we can/should write tests for a single method. Therefore, if we want to test the method “getCidrSizeFromString”, we should write a test case for that specific method, and not for another one that uses it. 

After we make sure that a method works fine, when we test another method (let’s say “cidrToLong”)  that uses the first one already tested, we can use EasyMock/Mockito or another test suite to mock the method that we know works, this way we facilitate the writing of test cases.